main.go

```diff
@@ -219,28 +226,60 @@ func buildProxy(host, bind *url.URL, tokenSource oauth2.TokenSource) *httputil.R
 	return proxy
 }
 
-// findTokenSource fetches the reusable/cached oauth2 token source. If t is
-// provided, that token is used as a static value. Othwerise, this attempts to
-// get the renewable token from the environment (including via Application
-// Default Credentials).
-func findTokenSource(t string) (oauth2.TokenSource, error) {
+// findTokenSource fetches the reusable/cached oauth2 token source. If rawToken
+// is provided, that token is used as a static value and the audience parameter
+// is ignored. Othwerise, this attempts to get the renewable token from the
+// environment (via Application Default Credentials).
+func findTokenSource(ctx context.Context, rawToken, audience string) (oauth2.TokenSource, error) {
 	// Prefer supplied value, usually from the flag.
-	if t != "" {
-		token := new(oauth2.Token).WithExtra(map[string]interface{}{
-			"id_token": t,
-		})
+	if rawToken != "" {
+		token := &oauth2.Token{AccessToken: rawToken}
 		return oauth2.StaticTokenSource(token), nil
 	}
 
-	// Try and find the default token from ADC.
-	ctx := context.Background()
-	tokenSource, err := google.DefaultTokenSource(ctx, cloudPlatformScope)
+	// Try to use the idtoken package, which will use the metadata service.
+	// However, the idtoken package does not work with gcloud, so we need to
```

Comment:
> the idtoken package does not work with gcloud

What does this mean? How is gcloud involved?

Reply:
The idtoken package doesn't work with gcloud's ADC. It specifically rejects "authorized_user" credentials. There's an [open issue](https://github.com/googleapis/google-api-go-client/issues/873) about it.

Reply:
Could you update the comment to include "gcloud's ADC"?

Reply:
Done